File: example/src/main/java/thirtytwo/degrees/halfpipe/example/view/ViewConfig.java
package thirtytwo.degrees.halfpipe.example.view;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
import thirtytwo.degrees.halfpipe.config.AbstractViewConfig;
import thirtytwo.degrees.halfpipe.config.MetricsConfig;
import thirtytwo.degrees.halfpipe.mgmt.view.MgmtControllers;

/**
 * User: spencergibb
 * Date: 9/21/12
 * Time: 4:18 PM
 * <p/>
 * http://static.springsource.org/spring/docs/3.1.x/spring-framework-reference/html/mvc.html#mvc-config-enable
 */
@EnableWebMvc
@Configuration
@ComponentScan(basePackageClasses = {ViewConfig.class, MgmtControllers.class}, excludeFilters = {
        @ComponentScan.Filter(Configuration.class)
})
@Import({MetricsConfig.class})
public class ViewConfig extends AbstractViewConfig {
    //TODO: is this needed, can I set a welcome page?
    @Override
    public void addViewControllers(ViewControllerRegistry registry) {
        registry.addViewController("/").setViewName("forward:/index.html");
    }
}

File: scala_2.9.1/src/main/scala/thirtytwo/degrees/halfpipe/scalaexample/view/ViewConfig.scala
package thirtytwo.degrees.halfpipe.scalaexample.view

import org.springframework.web.servlet.config.annotation.{ResourceHandlerRegistry, WebMvcConfigurerAdapter, EnableWebMvc}
import org.springframework.context.annotation.{ComponentScan, Configuration}
import thirtytwo.degrees.halfpipe.web.admin.AdminControllers

/**
 * User: spencergibb
 * Date: 9/22/12
 * Time: 4:11 PM
 */
@EnableWebMvc
@Configuration
@ComponentScan(basePackageClasses = Array(classOf[ViewConfig], classOf[AdminControllers]))
class ViewConfig extends WebMvcConfigurerAdapter {

  override def addResourceHandlers(registry: ResourceHandlerRegistry) {
    registry.addResourceHandler("/**").addResourceLocations("classpath:/static/")
  }
}
